### DISCUSSION TITLE
Monitor outputs

### DISCUSSION THREAD

**Discussion by juddmehr** [byuflowlab/FLOWUnsteady#30] [Q&A] [answered]
I'm currently saving forces on wings in custom frames via the `calc_aerodynamicforce()` and `decompose()` functions in FLOWUnsteady (combined with `_addsolution()` from FLOWVLM).  I need to do a similar thing, but with velocities on the wings.  Are there functions already in place that allow for that?

**Comment by EdoAlvarezR** [comment, marked as answer]
If I understand correctly, you are trying to obtain the effective velocity along the wing and decompose into arbitrary direction, right?

For debugging purposes, I exposed the effective velocities used in the force calculation in `calc_aerodynamicforce(...)` through the optional argument `Vout`. Notice in [Line 81](https://github.com/byuflowlab/FLOWUnsteady/blob/master/src/FLOWUnsteady_processing.jl) that if `Vout` is different than `nothing`, it will push to it an array with the vectors of the VPM-induced velocity (wake and embedded particles), VLM-induced velocity, freestream, and local kinematic velocity, all evaluated at the mid-point of the three bound vortices that make each horseshoe (TE to 1/4 chord, lifting line at 1/4 chord, and 1/4 chord to TE).

Hence, unfolding the two `for` loops around Line 81, `Vout[j][k]` is a 3-element array corresponding to one of the velocities (`k==1` => VPM-induced, `k==2` => VLM-induced, `k==3` => Vinf, `k==4` => Vkinematic), evaluated at the mid point of  the `l`-th bound vortex of the `i`-th horseshoe of `vlm_system`, where `l = (j-1)%3 + 1` and `i = floor(Int, (j-1)/3 + 1)`.

For example, to get the total velocity that was used to calculate the aerodynamic force at each segment of the lifting line:

```julia
Veff = []

for j in 1:3*vlm.get_m(vlm_system)

    l = (j-1)%3 + 1

    if l == 2
         push!(Veff, Vout[j][1] .+ Vout[j][2] .+ Vout[j][3] .+ Vout[j][4])
    end

end
```
Then you can use `decompose(...)` to decompose all the vectors into a different coordinate system.

**Reply by juddmehr** [reply]
Just a note on your example loop above: In a monitor where you pass in the simulation object, you're going to want to use sim.vehicle.vlm_system, unless you've already assigned that as the variable `vlm_system` in the extra runtime function.
